File: ocaml/bob/bob.ml
open Str

let isQuestion q =
	Str.string_match (Str.regexp ".+\?$") q 0

let isUpper q =
	String.equal (String.uppercase_ascii q) q

let hasLetters q =
	Str.string_match (Str.regexp ".+[a-zA-Z]+") q 0 ;;

let isYell q =
	hasLetters q && isUpper q

let response_for q =
	match (String.trim q) with
	| "" -> "Fine. Be that way!"
	| q when isYell q -> "Whoa, chill out!" 
	| q when isQuestion q -> "Sure."
	| _ -> "Whatever."

File: fsharp/bob/Bob.fs
module Bob

open System.Text.RegularExpressions

let isYell input =
    Regex.IsMatch(input, @"[A-Za-z]+") 
    && input.ToUpper() = input

let isQuestion input =
    Regex.IsMatch(input, @".+\?\z")

let hey (input:string) =
    let trimmed = input.Trim()
    match trimmed with
    | t when t = "" -> "Fine. Be that way!"
    | t when isYell t -> "Whoa, chill out!"
    | t when isQuestion t -> "Sure."
    | _ -> "Whatever."



